src/protocol/attribution/accumulate_credit.rs

```diff
@@ -63,12 +63,17 @@ impl<'a, F: Field> AccumulateCredit<'a, F> {
         // These vectors are updated in each iteration to help accumulate values and determine when to stop accumulating.
 
         let one = Replicated::one(ctx.role());
-        let mut stop_bits: Batch<Replicated<F>> = vec![one; num_rows as usize].try_into().unwrap();
+        // TODO - don't clone the value of one value so much
+        let mut stop_bits: Batch<Replicated<F>> = repeat(one.clone())
```

Comment:
This function is probably the most troubling as far as copying stuff around goes.  I wasn't able to do a major restructure here, so @taikiy probably has some work to do here.